frontend/src/metabase/query_builder/components/view/View.jsx

```diff
@@ -382,7 +388,14 @@ class View extends Component {
           </StyledDebouncedFrame>
         )}
 
-        {ModeFooter && (
+        {/* ModeFooter relies on `getMode` which relies on `qb.lastRunCard`,
+         so after a question modification, unless the result is refreshed,
+         we couldn't be sure if `mode` is still valid.
+         
+         We also couldn't just use `isResultDirty` because it doesn't consider
+         converting a question to a model as dirty.
+         */}
+        {ModeFooter && !isResultDirtyIgnoringModels && (
```

Comment:
The problem was that `isResultDirty` would still be `false` even if we just converted a question to a model.

But in this case of a footer, we want to make sure turning a question into a model is considered `dirty` since that changes the underlying query and [we don't really support models](https://github.com/metabase/metabase/blob/c47e63c34d9f37f01af91a76b0246a9075ec98ff/frontend/src/metabase/visualizations/click-actions/Mode/utils.ts#L15) when calculating mode in the first place.

Reply:
I don't understand how this change is connected to the original issue. Could you explain? If that's a separate issue please exclude it from the PR. It is very tricky and most likely incorrect.

Reply:
@ranquild 
It's true that it's a separate issue, but the fix I made crashed a lot of tests because we're currently showing the footer when we're not supposed to. And the fix assumes we would be able to get the question breakout and it's dimension which won't be the case when we show the footer when we're not supposed to.

I could make the fix more flexible with a lot of guard clauses (on `TimeseriesModeFooter.tsx`) to account for when we can't get question breakouts and dimensions, but that doesn't feel like the right thing to do since we know we're not supposed to also render the footer in some cases anyway. Then create a new PR to fix the issue where we render the footer when we're not supposed to then come back to remove all the guard clauses on `TimeseriesModeFooter.tsx`.

I don't like the fact that `master` would be in an intermediate state where the fix needs to have a lot of guard clauses because we still render it when we're not supposed to, but if that's best then I would do that.

what do you think? Should I extract not rendering the footer into another PR?

Reply:
Which tests exactly?

Reply:
- `e2e/test/scenarios/native/reproductions/32121-sql-source-query-can-explore-results.cy.spec.js`
- `e2e/test/scenarios/question/reproductions/28599-model-conversion-time-footer.cy.spec.js`
- `e2e/test/scenarios/models/models-with-aggregation-and-breakout.cy.spec.js`
- `e2e/test/scenarios/native/native.cy.spec.js`
- `e2e/test/scenarios/visualizations/reproductions/21615-convert-to-sql.cy.spec.js`

These would be all I think.

Reply:
The result can be seen from [this comment](https://github.com/metabase/metabase/pull/33810#issuecomment-1710507209) from deploysentinel and in its history

Reply:
@ranquild This [was an example](https://www.deploysentinel.com/ci/runs/64fb31bdcc8487467c779669), where the test crashed. That run was from before `isResultDirtyIgnoringModels` is introduced.

Reply:
@WiNloSt did you figure out why the app crashes?

Reply:
Yeah, basically it's because of the new fix as stated.
> It's true that it's a separate issue, but the fix I made crashed a lot of tests because we're currently showing the footer when we're not supposed to. And the fix assumes we would be able to get the question breakout and it's dimension which won't be the case when we show the footer when we're not supposed to.

It's basically from this few lines of the fix
https://github.com/metabase/metabase/blob/c2bb9e95f165421d8310ff9e8e7d6707c350d310/frontend/src/metabase/visualizations/click-actions/components/TimeseriesModeFooter.tsx#L16-L19

The fix assumes that it'd only be rendered when the question is timeseries. However that's not always the case e.g. when modifying a GUI query without rerunning the result, turning GUI question native, turning a GUI question into a model.

> I could make the fix more flexible with a lot of guard clauses (on TimeseriesModeFooter.tsx) to account for when we can't get question breakouts and dimensions, but that doesn't feel like the right thing to do since we know we're not supposed to also render the footer in some cases anyway.

I could also do that but it looks wrong in my opinion since the timeseries footer shouldn't have to care if the question passed isn't timeseries (it should always get timeseries question), but I'll check with the product first as you suggested.

Reply:
@WiNloSt take a look at these checks https://github.com/metabase/metabase/blob/31e83ed96dbe8d26b81a62db25ee841b5f6b163b/frontend/src/metabase/visualizations/click-actions/components/TimeseriesGroupingWidget.jsx#L22. I think we need to lift them up

Reply:
E.g. don't render `TimeseriesGroupingWidget` if it isn't a MBQL question or there are no breakouts, or if there are no subdimensions https://github.com/metabase/metabase/blob/1f7e261ec4702b36aa1f815440396187c4b7e931/frontend/src/metabase/query_builder/components/TimeGroupingPopover.jsx#L27

Reply:
@ranquild That would also work, but I think the real question is should `TimeseriesModeFooter.tsx` still need accept non-timeseries questions?

From these lines of code
https://github.com/metabase/metabase/blob/7e0eb68fd75fe69849dd263ce07dd2cb1aa67870/frontend/src/metabase/visualizations/click-actions/lib/modes.ts#L44-L45

It's where we determine what mode we should be using for the visualization. It feels to me that we should be able to expect that `TimeseriesMode.ModeFooter` would only get timeseries questions since it's been checked before.
https://github.com/metabase/metabase/blob/7e0eb68fd75fe69849dd263ce07dd2cb1aa67870/frontend/src/metabase/visualizations/click-actions/lib/modes.ts#L29

However, there is currently a problem with how this `getMode` function is being used in the selector.

https://github.com/metabase/metabase/blob/7e0eb68fd75fe69849dd263ce07dd2cb1aa67870/frontend/src/metabase/query_builder/selectors.js#L533-L538

The question is actually from the last run question, not the current question. That's why it's possible for the mode to be stale. e.g. when turning a question into a native question. We don't rerun the question, so `mode` would still be `timeseries` when it should have been `native` instead.

tl;dr is stale `mode` correct? I'm not sure about all the implications, but if it's correct then I could modify this fix so that the widget will account for non-timeseries questions passed to it.